src/status_im/subs/profile.cljs

```diff
@@ -325,6 +325,23 @@
  (fn [[profile ens-names port font-file] [_ avatar-opts]]
    (replace-multiaccount-image-uri profile ens-names port font-file avatar-opts)))
 
+(re-frame/reg-sub
+ :profile/profile-user-names
+ :<- [:profile/profile]
+ :<- [:ens/names]
+ (fn [[profile ens-names] _]
+   (let [display-name   (:display-name profile)
+         preferred-name (:preferred-name profile)]
+     (-> (map (fn [{:keys [name]}]
```

Comment:
Seeing a thread first with `map` is always a bit strange.

It reads better if you use `concat`, and then you will be able to use the thread last macro.

```clojure
(->> (range 10)
     (map inc)
     (concat [20]))
;; => insert 20 at the front (20 1 2 3 4 5 6 7 8 9 10) 
```

Reply:
Nice! Yeah that does feel nicer 🙌

Reply:
I think in this case I would disagree, `concat` is slower than `conj`, so I think if the reason is purely esthetical I don't think it's worth using `concat`, as well since if you are appending a single element, `concat` is not the right function to use and might be misleading other devs

Reply:
Yeah I think if we use `conj` we avoid creating that temporary vector too, so I suppose that's also good.
Is there some precedence for perf vs readability in the code guides or something?

Reply:
I disagree with the focus on nano optimizations to justify the code style. The UI is slow because we do certain things that are orders of magnitude slower. Not everything adds up to be meaningful. For instance, we pass maps to functions all over that destructure (instead of positional args), we make calls to style functions (instead of inlining them), and the list could go forever about nano details. To me, readability triumphs over such minuscule things.

> concat is slower than conj

On the JVM, `concat` seems to be faster for a small input. And notice here I'm even allocating a vector in the hot path.

```clojure
(let [a [2]]
  (criterium/with-progress-reporting
    (criterium/bench
        (concat [1] a))))
;; Execution time mean : 3.931832 ns
;; Execution time std-deviation : 0.234408 ns

(let [a '(2)]
  (criterium/with-progress-reporting
    (criterium/bench
        (conj a 1))))
;; Execution time mean : 4.601176 ns
;; Execution time std-deviation : 0.215105 ns
``` 

We can't really trust `simple-benchmark` results, but `conj` *can be* slower in CLJS.

```clojure
(simple-benchmark [a [2]] (concat [1] a) 1000000) ; 1757 msecs
(simple-benchmark [a [2]] (conj a 1) 1000000) ; 2218 msecs
```

> concat is not the right function to use and might be misleading other devs

Devs should be familiar with `concat`, it's in the core lib. `concat` is more obvious for certain devs I worked with, because the order of the arguments is respected in the final result, whereas with `conj`, if you always want to insert in the front, you need to know if the data structure is not a vector, which sometimes is hard because the data was passed down. I've seen bugs due to this.

Thread first with `map` made me process the code more to understand it because it's uncommon, but I'm sure it's a matter of preference. But saying `concat` is not the right function is a bit excessive.

Reply:
"I disagree with the focus on nano optimizations to justify the code style. " 

That's not quite what it is, as I mentioned, I think `conj` is the right function if you are adding an element to collection, in this case it prepends which is exactly what we want. `concat` is for concatenating two lists and if you are using it instead of `conj`, it communicates that the intention of the developer is different. It's a problem of expressiveness rather than strictly performance, and I think that a more junior dev could get the wrong idea seeing code like this.

Regarding benchmark, I am sure you know about the pitfalls of benchmarking lazy functions such as `concat` ;)